### PR TITLE
Test Go autobuild

### PR DESCRIPTION
**Pull request by robertbrignull**
We should be testing using autobuild with Go as well as a manual build command, just in case.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/master/README.md) has been updated if necessary.

### CODE REVIEW

**Comment by robertbrignull** [comment]
I wanted to make sure the Go tracing was working if you used the autobuild action and thought we may as well add an integration test for it. Turns out it does work.

File: .github/workflows/integration-testing.yml [hunk 1]
```diff
@@ -129,6 +129,33 @@ jobs:
...
+    # we're testing Go custom tracing with a manual build on all OSes.
+    runs-on: ubuntu-latest
+    env:
+      CODEQL_EXTRACTOR_GO_BUILD_TRACING: "on"
+
+    steps:
+    - uses: actions/setup-go@v2
+      if: ${{ matrix.os ==  'macos-latest' }}
+      with:
+        go-version: '^1.13.1'
```
Comment (chrisgavin) [inline .github/workflows/integration-testing.yml:143]:
```suggestion
```

Does this do anything? It looks like the job always runs on Ubuntu so this step will always be skipped.

**Review by chrisgavin — APPROVED** [review]
Looks good with the exception of one question.

File: .github/workflows/integration-testing.yml (hunk 1 above)
Comment (robertbrignull) [reply .github/workflows/integration-testing.yml:143]: Thanks. I accidentally copied that from the other job.

**Pull request merged**